Packs/Nexthink/Integrations/Nexthink/Nexthink.py

```diff
@@ -0,0 +1,187 @@
+import json
+import urllib.parse
+
+import demistomock as demisto  # noqa: F401
+import requests
+from CommonServerPython import *  # noqa: F401
+
+# disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+entryList = []
+deviceList = []
+deviceEntry = {}
+
+args = demisto.args()
+
+if 'hostname' in args:
+    device = args['hostname']
+else:
+    device = None
+
+if 'ipaddress' in args:
+    ip = args['ipaddress']
+else:
+    ip = None
+
+if 'package' in args:
+    package = args['package']
+else:
+    package = None
+
+SEARCH_DEVICE_USING_IP = "(select (*) (from device (where device ( eq ip_addresses (ip_address '{0}')))))".format(ip)
+SEARCH_DEVICE_USING_DEVICE = "(select (*) (from device (where device ( eq name (string {0})))))".format(device)
+SEARCH_COMPLIANCE_PACKAGE_DEVICE = """(select ((device (*)) (package (*))) (from (device package)
+(with package (where package (eq name (pattern '*{0}*')))
+(where device (eq name (pattern '{1}')))))
+(limit 100))""".format(package, device)
+TEST_MODULE = "(select (name) (from device ) (limit 1))"
+
+
+def nexthink_request(method, nxql):
+    params = demisto.params()
+    username = params.get('credentials').get('identifier')
+    password = params.get('credentials').get('password')
+    base_url = params.get('url')
+    port = params.get('port')
+    verify_ssl = not params.get('insecure', False)
+    proxy = params.get('proxy', False)
+
+    if proxy:
+        proxies = handle_proxy()
+    else:
+        proxies = {
+            "http": None,
+            "https": None,
+        }
+
+    BASE_URL = 'https://{0}:{1}/2/query?platform=windows&format=json&query='.format(base_url, port)
+    NXQL = urllib.parse.quote(nxql)
+    urlFragment = BASE_URL + NXQL
+
+    try:
+        if method == 'POST':
+            response = requests.post(urlFragment, auth=(username, password), verify=verify_ssl, proxies=proxies)
+        else:
+            response = requests.get(urlFragment, auth=(username, password), verify=verify_ssl, proxies=proxies)
+
+        if response.status_code == 200:
+            return json.loads(response.content)
+        else:
+            return str(response.status_code)
+    except requests.Timeout:
+        raise
+    except requests.ConnectionError:
+        raise
```

Comment:
Consider adding a message about the failure so the user would know why was the command failed.